### PR TITLE
Consume configuration system from Arcade

### PR DESCRIPTION
**Pull request by ericstj**
Currently I'm consuming a private build of my arcade package until that portion is merged.

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -26,7 +26,44 @@
...
+  <!-- Import configuration data model -->
+  <Import Project="$(RepositoryEngineeringDir)configurations/properties.props" />
+  
+  <PropertyGroup Condition="'$(BuildingInsideVisualStudio)' == 'true' and '$(Configuration)' != ''">
+    <!-- When building in VS setup the ConfigurationGroup based on the given Configuration -->
+    <ConfigurationGroup Condition="$(Configuration.EndsWith('Debug'))">Debug</ConfigurationGroup>
+    <ConfigurationGroup Condition="$(Configuration.EndsWith('Release'))">Release</ConfigurationGroup>
+  </PropertyGroup>
+  
+  <!-- Platform detection -->
```
Comment (ericstj) [inline Directory.Build.props]: I moved this from BuildTools.

File: Directory.Build.props [hunk 2]
```diff
@@ -26,7 +26,44 @@
...
+  <PropertyGroup>
+    <DefaultOSGroup Condition="'$(OS)'=='Unix' AND Exists('/Applications')">OSX</DefaultOSGroup>
+    <DefaultOSGroup Condition="'$(DefaultOSGroup)'=='' AND '$(OS)'=='Unix' AND $([MSBuild]::IsOSPlatform('FREEBSD'))">FreeBSD</DefaultOSGroup>
+    <DefaultOSGroup Condition="'$(DefaultOSGroup)'=='' AND '$(OS)'=='Unix' AND $([MSBuild]::IsOSPlatform('NETBSD'))">NetBSD</DefaultOSGroup>
+    <DefaultOSGroup Condition="'$(DefaultOSGroup)'=='' AND '$(OS)'=='Unix'">Linux</DefaultOSGroup>
+    <DefaultOSGroup Condition="'$(DefaultOSGroup)'==''">$(OS)</DefaultOSGroup>
+  </PropertyGroup>
+
+  <PropertyGroup>
```
Comment (ericstj) [inline Directory.Build.props]: This was merely moved from lower down in this same file

File: Directory.Build.props [hunk 3]
```diff
@@ -57,6 +101,9 @@
     <SourceDir>$(ProjectDir)src\</SourceDir>
     <BinDir>$(ProjectDir)artifacts\bin\</BinDir>
 
+    <!-- Need to try and keep the same logic as the native builds as we need this for packaging -->
+    <NativeBinDir>$(BinDir)native/$(BuildConfiguration)</NativeBinDir>
```
Comment (ViktorHofer) [inline Directory.Build.props]: Not 100% sure if you really want to rely on $(BinDir). Wes grouped it to other obsolete properties which should be removed when buildtools is gone.

Comment (safern) [reply Directory.Build.props]: Arcade defines: `ArtifactsBinDir` https://github.com/dotnet/arcade/blob/a991a1d46cf6e5ed7dde93309217306f6a42c136/src/Microsoft.DotNet.Arcade.Sdk/tools/RepoLayout.props#L68

But the SDK props are not imported until ~ line 200

Comment (ericstj) [reply Directory.Build.props]: This isn't some new dependency, it's just reordering an existing property.  I'm ok with it moving to a different path / property but I don't think I should be doing that in this PR.

Comment (ViktorHofer) [reply Directory.Build.props]: Agreed, let's defer it till we remove buildtools entirely.

**Comment by ahsonkhan** [comment]
@dotnet-bot test this please

**Comment by ericstj** [comment]
Test NETFX x86 Release Build

**Comment by ericstj** [comment]
Test Windows x86 Release Build

**Comment by safern** [comment]
test Windows x64 Debug Build
test Windows x86 Release Build

Queue was full.

**Comment by ericstj** [comment]
Conflict... :(  I'll wait for CI to finish, fix the conflict and push without CI.  /cc @ViktorHofer

**Comment by ViktorHofer** [comment]
@ericstj did that for you as CI just finished when I looked at it.

**Comment by ericstj** [comment]
ha, you beat me by a minute.  I was scratch my head as I tired to push and it said I needed to merge the remote.

**Comment by ViktorHofer** [comment]
😁. Usually I just use the web UI to merge such simple conflicts.

**Pull request merged**